src/RpcClient.ts

```diff
@@ -354,4 +372,119 @@ export class RpcClient {
+        await initTree(this.connection, ownerKeypair, treeWallet);
+      }
+      let response: any;
+      let collection = request.collection ?? false;
+
+      // User set collection to false, or it was not dictated.
+      if (collection == false && treeWallet) {
+        const signature = await mintCompressedNFT(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet
+        );
+        const { id } = await getAssetID(
+          this.connection.rpcEndpoint,
+          treeWallet.publicKey,
+          signature
+        );
+        response = {
+          assetId: id,
+          creator: request.ownerKeypair.publicKey.toBase58(),
+          treeId: treeWallet.publicKey.toBase58(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+        assetId = id;
+        // User wants a collection
+      } else if (collection == true) {
+        let collectionMint = request.collectionMint;
+        // User did not supply a public key for exisiting collection
+        if (!collectionMint) {
+          const collectionInfo = await initCollection(
+            this.connection,
+            ownerKeypair,
+            request.metadataArgs.name,
+            request.metadataArgs.symbol,
+            request.metadataArgs.uri
+          );
+          collectionMint = collectionInfo.collectionMint;
+          // User did supply a collection mint Public key.
+        } else if (collectionMint) {
+          const { collectionMetadataAccount, collectionMasterEditionAccount } =
+            await existingCollection(collectionMint);
+          const signature = await mintCollectionCompressedNft(
+            this.connection,
+            defaultMetadataArgs,
+            request.ownerKeypair,
+            treeWallet,
+            collectionMint,
+            collectionMetadataAccount,
+            collectionMasterEditionAccount
+          );
+          // Get Merkle tree data + Asset ID after minting the NFT
+          const { id } = await getAssetID(
+            this.connection.rpcEndpoint,
+            treeWallet.publicKey,
+            signature
+          );
+          assetId = id;
+          response = {
+            assetId: id,
+            creator: request.ownerKeypair.publicKey.toBase58(),
+            treeId: treeWallet.publicKey.toBase58(),
+            treeKeypair: treeWallet.secretKey,
+            signature: signature,
+          };
+        }
+      }
+      let DAS;
+      let confirm = request.confirmMint ?? true;
+      if (confirm == true) {
+        while (true) {
+          try {
+            DAS = await this.getAsset(assetId.toString());
+
+            if (DAS) {
+              break;
+            }
+          } catch (e) {
+            new Error(
+              `Error in retrieving from DAS. Retrying in 1 second: ${e}`
+            );
+            await new Promise((resolve) => setTimeout(resolve, 500));
+          }
+        }
+      }
```

Comment:
We could probably limit this to a finite number of retries to avoid getting stuck in an infinite loop.